src/CMakeLists.txt

```diff
@@ -8,23 +8,21 @@ project(Rover LANGUAGES CXX C)
 find_package(OpenCV)
 find_package(Eigen3 REQUIRED NO_MODULE)
 find_package(Threads REQUIRED)
-
+find_package(Catch2 REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(geometry_msgs REQUIRED)
 include_directories(${rclcpp_INCLUDE_DIRS})
 include_directories(${geometry_msgs_INCLUDE_DIRS})
 
 add_definitions(-DCHIP_TYPE=CHIP_TYPE_TEMPLATE)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall \
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -Wall \
```

Comment:
I figure we might as well compile with debug symbols by default. Still runs plenty fast on my machine.

Reply:
I think there's actually a cmake way of doing this: https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

We should probably use that; I think we can just set it to "Debug" by default if it's not explicitly set from the command line. I'm reviewing this on the light rail lol so I'll provide a code snippet later.

Reply:
Ah, I see: it seems you can set `-DCMAKE_BUILD_TYPE=Debug` the first time you run cmake, and it'll keep that build type for all future calls to `make`. I hadn't quite understood how that worked. I'll revert this!